.github/workflows/test.yml

```diff
@@ -68,3 +71,26 @@ jobs:
       run: |
         cd e2e
         python -m pytest .
+
+    - name: Rust library tests
+      run: |
+        cd rust/rsmgp-sys
+        cargo fmt -- --check
+        cargo test
+    - name: Install Cargo tarpaulin # Unable to run https://github.com/actions-rs/tarpaulin from subdirectory.
+      run: |
+          cargo install cargo-tarpaulin
+    - name: Run code coverage
+      run: |
+        cd rust/rsmgp-sys
+        RUST_BACKTRACE=1 cargo tarpaulin --out Html --ignore-tests
+    - name: Archive code coverage results
+      uses: actions/upload-artifact@v1
+      with:
+        name: rsmgp-sys-coverage-report
+        path: rust/rsmgp-sys/tarpaulin-report.html
+    - name: Rust example build
+      run: |
+        cd rust/rsmgp-example
+        cargo fmt -- --check
+        cargo build
```

Comment:
Can you split this in two separate units - building and testing?

Reply:
What do you mean when you say "split"? `cargo test` also builds the code + here `rsmgp-sys` and `rsmgp-example` are built. Did you mean to just puts an empty line between `rsmgp-sys` and `rsmgp-example`?

Reply:
No, I meant that building the code should be inside `name: Set up Rust`, while running the test should be under `name: Rust library tests`. This is only for the convenience of someone who is looking in those actions on Github

Reply:
Does the current state make sense?